lib.php

```diff
@@ -980,3 +999,31 @@ function reengagement_checkstart($course, $cm, $reengagement) {
     }
     return $output;
 }
+
+/**
+ * Check that emailfrom user has capability to add reengagments,
+ *  otherwise return support user
+ *
+ * @param object $reengagement
+ * @return object $user
+ */
+function reengagement_get_emailfrom($reengagement) {
+    $userid = $reengagement->emailfrom;
+    if ($userid > 0) {
+        $context = context_course::instance($reengagement->course);
+        if ($userid == 1) { // Default teacher, get first teacher in course.
```

Comment:
better to use something like is siteadmin() style code - userid ===1 might not always be the siteadmin.

Reply:
This should be the variable REENGAGEMENT_EMAILFROM_TEACHER, but I take your point that the user with id 1 may not be the siteadmin, so I'm changing the value for default teacher to -1, 0 is still the support user, and then any integer above 0 will be the actual user.